Some Python 3.7,x versions do not seem to install well.

### What did you try to do?

Install pycurl when building Docker Image.

### What happened?

When the image is created by building the following Dockerfile, the install is failing.

```
# Dockerfile
# 3.7.4, 3.7.5, 3.7.7, 3.7.8, 3.7.9, 3.7.10
FROM python:3.7.6
RUN pip3 install pycurl
```

### What did you expect to happen?

The 7.44.1 version is installed normally, but after this release, I wonder if any part of the 7.45.0 version is not supported.

### What is the PycURL version?

7.45.0

### What is your Python version?

3.7.6

### What is your operating system and its version?

https://hub.docker.com/layers/python/library/python/3.7.6/images/sha256-23c8c0222fed7e88257767551ca7fdbcde09835197db57d5eb44d1f6bcd818b6?context=explore

### Other versions

doesn't matter

### Is this the most recent PycURL release?

yes

### Did you check libcurl behavior?

N/A

The following error occurs during installation.

'''
#4 5.336   src/module.c: In function ‘PyInit_pycurl’:
#4 5.336   src/module.c:379:22: error: ‘CURLSSLBACKEND_SECURETRANSPORT’ undeclared (first use in this function); did you mean ‘CURLSSLBACKEND_SCHANNEL’?
#4 5.336                    case CURLSSLBACKEND_SECURETRANSPORT:
#4 5.336                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#4 5.336                         CURLSSLBACKEND_SCHANNEL
#4 5.336   src/module.c:379:22: note: each undeclared identifier is reported only once for each function it appears in
#4 5.336   error: command 'gcc' failed with exit status 1
#4 5.336   ----------------------------------------
#4 5.336   ERROR: Failed building wheel for pycurl
#4 5.337   Running setup.py clean for pycurl
#4 5.558 Failed to build pycurl
#4 5.587 Installing collected packages: pycurl
#4 5.588     Running setup.py install for pycurl: started
#4 7.684     Running setup.py install for pycurl: finished with status 'error'
#4 7.686     ERROR: Command errored out with exit status 1:
#4 7.686      command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-k91f83nm/pycurl/setup.py'"'"'; __file__='"'"'/tmp/pip-install-k91f83nm/pycurl/setup.py'"'"';f=getattr(tokenize, '"
'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-rv29zrpo/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.7m/pycurl
#4 7.686          cwd: /tmp/pip-install-k91f83nm/pycurl/
#4 7.686     Complete output (27 lines):
#4 7.686     Using curl-config (libcurl 7.64.0)
#4 7.686     Using SSL library: OpenSSL/LibreSSL/BoringSSL
#4 7.686     running install
#4 7.686     running build
#4 7.686     running build_py
#4 7.686     creating build
#4 7.686     creating build/lib.linux-x86_64-3.7
#4 7.686     creating build/lib.linux-x86_64-3.7/curl
#4 7.686     copying python/curl/__init__.py -> build/lib.linux-x86_64-3.7/curl
#4 7.686     running build_ext
#4 7.686     building 'pycurl' extension
#4 7.686     creating build/temp.linux-x86_64-3.7
#4 7.686     creating build/temp.linux-x86_64-3.7/src
#4 7.686     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYCURL_VERSION="7.45.0" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/usr/local/include/python3.7m -c src/docstrings.c -o build/temp.linux-x86_64-3.7/src/docstrings.o
#4 7.686     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYCURL_VERSION="7.45.0" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/usr/local/include/python3.7m -c src/easy.c -o build/temp.linux-x86_64-3.7/src/easy.o
#4 7.686     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYCURL_VERSION="7.45.0" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/usr/local/include/python3.7m -c src/easycb.c -o build/temp.linux-x86_64-3.7/src/easycb.o
#4 7.686     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYCURL_VERSION="7.45.0" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/usr/local/include/python3.7m -c src/easyinfo.c -o build/temp.linux-x86_64-3.7/src/easyinfo.o
#4 7.686     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYCURL_VERSION="7.45.0" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/usr/local/include/python3.7m -c src/easyopt.c -o build/temp.linux-x86_64-3.7/src/easyopt.o
#4 7.686     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYCURL_VERSION="7.45.0" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/usr/local/include/python3.7m -c src/easyperform.c -o build/temp.linux-x86_64-3.7/src/easyperform.o
#4 7.686     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYCURL_VERSION="7.45.0" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/usr/local/include/python3.7m -c src/module.c -o build/temp.linux-x86_64-3.7/src/module.o
#4 7.686     src/module.c: In function ‘PyInit_pycurl’:
#4 7.686     src/module.c:379:22: error: ‘CURLSSLBACKEND_SECURETRANSPORT’ undeclared (first use in this function); did you mean ‘CURLSSLBACKEND_SCHANNEL’?
#4 7.686                      case CURLSSLBACKEND_SECURETRANSPORT:
#4 7.686                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#4 7.686                           CURLSSLBACKEND_SCHANNEL
#4 7.686     src/module.c:379:22: note: each undeclared identifier is reported only once for each function it appears in
#4 7.686     error: command 'gcc' failed with exit status 1
#4 7.686     ----------------------------------------
#4 7.697 ERROR: Command errored out with exit status 1: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-k91f83nm/pycurl/setup.py'"'"'; __file__='"'"'/tmp/pip-install-k91f83nm/pycurl/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-rv29zrpo/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.7m/pycurl Check the logs for full command output.

'''

--- a/src/module.c
+++ b/src/module.c
@@ -376,7 +376,11 @@ initpycurl(void)
                 case CURLSSLBACKEND_NSS:
                 case CURLSSLBACKEND_WOLFSSL:
                 case CURLSSLBACKEND_MBEDTLS:
+#if LIBCURL_VERSION_NUM >= MAKE_LIBCURL_VERSION(7, 64, 1)
                 case CURLSSLBACKEND_SECURETRANSPORT:
+#else
+                case CURLSSLBACKEND_DARWINSSL:
+#endif
                     runtime_supported_backend_found = 1;
                     break;
                 default:

